### PR TITLE
Added more tests for Franz Integration

### PR DESCRIPTION
**Pull request by marian-craciunescu**

### CODE REVIEW

File: server/router/router.go [hunk 1]
```diff
@@ -188,6 +190,29 @@ func (router *router) HandleMessage(message *protocol.Message) error {
 	return nil
 }
 
+func (router *router) GetSubscribersForTopic(topicPath string) (JSONResponse []byte, err error) {
```
Comment (Bogh) [inline server/router/router.go]: Remove the returned variables or use them

File: server/router/router.go [hunk 2]
```diff
@@ -188,6 +190,29 @@ func (router *router) HandleMessage(message *protocol.Message) error {
...
+
+	sliceOfRouteParams := make([]map[string]string, 0)
```
Comment (Bogh) [inline server/router/router.go]: Rename to subscribers and use `RouteParams` as slice data.

File: server/router/router.go [hunk 3]
```diff
@@ -188,6 +190,29 @@ func (router *router) HandleMessage(message *protocol.Message) error {
...
+
+	slice, present := router.routes[protocol.Path(topicPath)]
```
Comment (Bogh) [inline server/router/router.go]: Rename `slice` to `routes`

File: server/router/router.go [hunk 4]
```diff
@@ -188,6 +190,29 @@ func (router *router) HandleMessage(message *protocol.Message) error {
...
+	if present {
+		for index, currRoute := range slice {
+			logger.WithFields(log.Fields{
+				"index":      index,
+				"routeParam": currRoute.RouteParams,
+			}).Info("Added route to slice")
+			sliceOfRouteParams = append(sliceOfRouteParams, currRoute.RouteParams)
+		}
+		serialized, err := json.Marshal(sliceOfRouteParams)
+		logger.WithField("json_Response", string(serialized)).Info("Serialized json as string")
```
Comment (Bogh) [inline server/router/router.go]: Dont use Info logging for this. Can be quite big data.

File: server/router/router.go [hunk 5]
```diff
@@ -188,6 +190,29 @@ func (router *router) HandleMessage(message *protocol.Message) error {
...
+		if err != nil {
+			return nil, err
+		}
+		return serialized, nil
+	}
+	return nil, ErrInvalidRoute
```
Comment (Bogh) [inline server/router/router.go]: Why return invalid route in this case. If topic is not found better return an empty data.
Don't think it's needed to error. Maybe no one subscribed in that topic which makes us ok with that.

This method should always return an empty serialized slice.

File: server/rest/rest_message_api.go [hunk 6]
```diff
@@ -48,6 +50,27 @@ func (api *RestMessageAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		log.WithField("topic", topic).WithField("err", err).Info("Extract")
+		if err != nil {
+			if err == errNotFound {
+				http.NotFound(w, r)
+				return
+			}
+			http.Error(w, "Server error.", http.StatusInternalServerError)
+			return
+		}
+		resp, err := api.router.GetSubscribersForTopic(topic)
```
Comment (Bogh) [inline server/rest/rest_message_api.go]: Error not treated.

File: server/rest/rest_message_api.go [hunk 7]
```diff
@@ -48,6 +50,27 @@ func (api *RestMessageAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if r.Method == http.MethodGet {
+		log.WithField("url", r.URL.Path).Info("Get ")
+		topic, err := api.extractTopic(r.URL.Path, subscribersPrefix)
+		log.WithField("topic", topic).WithField("err", err).Info("Extract")
+		if err != nil {
```
Comment (Bogh) [inline server/rest/rest_message_api.go]: Log the error please.

File: server/integration_test.go [hunk 8]
```diff
@@ -94,3 +102,61 @@ func checkConnectedNotificationJSON(t *testing.T, user string, connectedJSON str
...
+	a.Equal(4, len(routeParams), "Should have 4 subscribers")
+	for i, rp := range routeParams {
+		a.Equal(fmt.Sprintf("gcmId%d", i), rp.DeviceID)
+		a.Equal(fmt.Sprintf("%d", i), rp.UserID)
+	}
+	a.NoError(err)
+
+}
+
+func subscribe_multiple_clients(t *testing.T, service *service.Service, noOfClients int) {
```
Comment (Bogh) [inline server/integration_test.go]: camelCase method name please.

File: server/router/router.go (hunk 1 above)
Comment (marian-craciunescu) [inline server/router/router.go]: Done

File: server/router/router.go (hunk 2 above)
Comment (marian-craciunescu) [inline server/router/router.go]: done

File: server/router/router.go (hunk 3 above)
Comment (marian-craciunescu) [inline server/router/router.go]: done

File: server/router/router.go (hunk 4 above)
Comment (marian-craciunescu) [inline server/router/router.go]: DONE

File: server/router/router.go (hunk 5 above)
Comment (marian-craciunescu) [inline server/router/router.go]: Done...IT WILL return if empty a `[]`

File: server/rest/rest_message_api.go (hunk 6 above)
Comment (marian-craciunescu) [inline server/rest/rest_message_api.go]: done

File: server/integration_test.go (hunk 8 above)
Comment (marian-craciunescu) [inline server/integration_test.go]: done

**Pull request merged**